src/bsp/platform/esp32/xi_bsp_io_fs_esp32.c

```diff
@@ -0,0 +1,307 @@
+/* Copyright (c) 2003-2017, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client library,
+ * it is licensed under the BSD 3-Clause license.
+ */
+
+#include <sys/stat.h>
+#include <errno.h>
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <xi_bsp_debug.h>
+#include <xi_bsp_io_fs.h>
+#include <xi_bsp_fwu.h>
+#include <xi_bsp_mem.h>
+
+#include "esp_vfs.h"
+#include "esp_vfs_fat.h"
+#include "esp_ota_ops.h"
+
+#define XI_ESP32_MAX_FILENAME_LEN 64
+#define ESP32_FS_BASE_PATH "/spiflash"
+
+#define XI_BSP_IO_FS_CHECK_CND( cnd, e, s )                                              \
+    if ( ( cnd ) )                                                                       \
+    {                                                                                    \
+        ( s ) = ( e );                                                                   \
+        perror( "Xi FS BSP errno" );                                                     \
+        goto err_handling;                                                               \
+    }
+
+/* These functions are implemented in the application layer to report status */
+extern void esp32_xibsp_notify_update_started( const char* filename, size_t file_size );
+extern void esp32_xibsp_notify_chunk_written( size_t chunk_size, size_t offset );
+
+/* The size of the buffer to be used for reads. */
+const size_t xi_bsp_io_fs_buffer_size = 1024;
+/* ESP32 Over The Air FW updates API handle */
+static esp_ota_handle_t open_firmware_bin_handle = 0;
+
+/* translates bsp errno errors to the xi_bsp_io_fs_state_t values */
+xi_bsp_io_fs_state_t xi_bsp_io_fs_posix_errno_2_xi_bsp_io_fs_state( int errno_value )
```

Comment:
..._posix_errno_2_... does the ESP32 FAT API return posix error values? Is the mapping valid for ESP32 too?
If yes, then ok. If not, then update the mapping and rename the function pls.

Reply:
I have not validated the map entries one by one, but yes, the ESP32 uses standard POSIX errno values.

Reply:
ok